### ISSUE TITLE
Nested Timestamp shown as "Invalid Date"

### ISSUE THREAD

**Issue by dhstack** [benwinding/react-admin-firebase#108]
A timestamp field at the root document level works as expected with this line:
`<DateField source="submittedAt" showTime />`

With these results:
![image](https://user-images.githubusercontent.com/2013388/79670457-39e87c80-8191-11ea-96ec-5f56ebe590c5.png)
![image](https://user-images.githubusercontent.com/2013388/79670465-4240b780-8191-11ea-9055-3994653e3027.png)

But a timestamp field in a Map object (e.g. `mymap.timestamp`) is rendered as `Invalid Date` in the List view and the Show view with this line:
`<DateField source="submission.submittedAt" label="Submitted at" showTime />`

And these are the results:
![image](https://user-images.githubusercontent.com/2013388/79670412-c3e41580-8190-11ea-8996-6bd7beed2bfa.png)
![image](https://user-images.githubusercontent.com/2013388/79670420-dc543000-8190-11ea-9e29-473154c5db66.png)

**Comment by dhstack** [comment]
My current workaround is using a [`Function Field`](https://marmelab.com/react-admin/Fields.html#functionfield) instead of a `DateField` and directly getting the seconds and converting to milliseconds, but it's not ideal.

```
<FunctionField
  label="Submitted at"
  render={record =>
    record.submission.submittedAt
      ? new Date(
          record.submission.submittedAt.seconds * 1000
        ).toLocaleString()
      : 'N/A'
  }
/>
```

**Comment by benwinding** [comment]
Thanks for reporting @dhstack,

Should be fixed from PR #114 , and deployed as version `3.1.13`.

Basically, it wasn't converting Date fields recursively. But that should be fixed now, and is also checking this in the new unit-tests.

https://github.com/benwinding/react-admin-firebase/blob/b2011c4638d622fc0599eb22d76df5b4a745bc05/tests/integration-tests/ApiGetOne.spec.ts#L28-L52

Cheers,
Ben

**Comment by dhstack** [comment]
Works for me, thanks!
